src/dbms/dbms_handler.hpp

```diff
@@ -515,18 +628,35 @@ class DbmsHandler {
     }
     throw UnknownDatabaseException("Tried to retrieve an unknown database \"{}\".", name);
   }
+#endif
+
+  void RecoverReplication(DatabaseAccess db_acc) {
+    if (allow_mt_repl || db_acc->name() == dbms::kDefaultDB) {
+      // Handle global replication state
+      spdlog::info("Replication configuration will be stored and will be automatically restored in case of a crash.");
+      // RECOVER REPLICA CONNECTIONS
+      memgraph::dbms::RestoreReplication(repl_state_, std::move(db_acc));
+    } else if (const ::memgraph::replication::RoleMainData *data =
+                   std::get_if<::memgraph::replication::RoleMainData>(&repl_state_.ReplicationData());
+               data && !data->registered_replicas_.empty()) {
+      spdlog::warn("Multi-tenant replication is currently not supported!");
```

Comment:
TODO: we need to implement

Reply:
I was incorrect, this will be cleaned up once we remove the Experimental build flag.